### DISCUSSION TITLE
fromModule of packetArrived()

### DISCUSSION THREAD

**Discussion by ytrewq271828** [ANLAB-KAIST/KENSv3#164] [Project Q&A 2. Transport Layer (TCP)]
What does the fromModule argument in packetArrived() indicate?
I assume that this would be related to information regarding the sender process.
But I wasn't able to figure out what would be the exact value of the argument.
Is that the name of the sender process? Or something other than that?

Thanks in advance.

**Comment by Pusnow** [comment]
It is the name of sender element. In KENS, only "IPv4" is passed, so you can ignore the argument.

**Reply by ytrewq271828** [reply]
Then what about the second parameter &&packet? 
Is it the incoming packet from network layer? Then how should we call packetArrived() when implementing handshakes?

**Reply by Pusnow** [reply]
> Is it the incoming packet from network layer?

Yes, upon packet arrivals, the `packetArrived()` method is called by KENS.

> Then how should we call packetArrived() when implementing handshakes?

You can't. You only need to implement body of the method. To send packets, you should use `sendPacket()` method. See: https://github.com/ANLAB-KAIST/KENSv3/wiki/Tip:-Sending-and-Receiving-a-Packet
